workspaces/analytics/src/interfaces/RegisterEvent.ts

```diff
@@ -0,0 +1,50 @@
+import { TrackingEventBase } from './TrackingEventBase';
+// @ts-ignore
+import * as Joi from '@hapi/joi';
+import 'joi-extract-type';
+
+function RegisterEvent<T>(
+  type: string,
+  schema: any,
+  propsToSentence: (propsInstance: T) => string
+): RegisteredEvent<T> {
+  const event = {
+    withProps(properties: T): TrackingEventBase<T> {
+      return {
+        type: type,
+        context: {
+          clientAgent: '',
```

Comment:
Some of these we'll need to figure out where they come from. if all tracking is authenticated we can fill in clientId on the server side

Reply:
it stubs empty and each client sets these themselves